search/views.py

```diff
@@ -51,7 +51,7 @@ def do_search(request, course_id=None):
     }
     status_code = 500
 
-    search_terms = request.POST["search_string"]
+    search_terms = request.POST.get("search_string", None)
```

Comment:
Well, quite right, we should really short-circuit it here - it will be okay as None, but there is no point in executing a query in that case - I shall divert